### PR TITLE
Add Fingerprint API to IDX bundle

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
Currently `fingerprint` method is available from `authn` bundle. 
This PR adds this method to `idx` bundle (to be used in SIW, see [PR](https://github.com/okta/okta-signin-widget/pull/3680))

[OKTA-418160](https://oktainc.atlassian.net/browse/OKTA-418160)

### CODE REVIEW

**Comment by lesterchoi-okta** [comment]
IIRC this was blocked on release issues. @jaredperreault-okta is this going into an upcoming release?

File: lib/browser/fingerprint.ts [hunk 1]
```diff
@@ -17,31 +17,38 @@ import {
...
 
-  var timeout;
-  var iframe;
-  var listener;
-  var promise = new Promise(function (resolve, reject) {
+  const container = options?.container ?? document.body;
+  let timeout: NodeJS.Timeout;
+  let iframe: HTMLIFrameElement;
+  let listener: (this: Window, ev: MessageEvent) => void;
+  let msg;
```
Comment (jaredperreault-okta) [inline lib/browser/fingerprint.ts:35]: is `msg` used outside of `listener`? If not, we should move into the `listener` function

File: lib/browser/fingerprint.ts [hunk 2]
```diff
@@ -52,17 +59,18 @@ export default function fingerprint(sdk: OktaAuthHttpInterface, options?: Finger
...
         return resolve(msg.fingerprint as string);
-      }
-      if (msg.type === 'FingerprintServiceReady') {
-        e.source.postMessage(JSON.stringify({
+      } else if (msg.type === 'FingerprintServiceReady') {
+        iframe?.contentWindow?.postMessage(JSON.stringify({
           type: 'GetFingerprint'
         }), e.origin);
+      } else {
+        return reject(new Error('No data'));
```
Comment (jaredperreault-okta) [inline lib/browser/fingerprint.ts:67]: Should this be an `AuthSDKError`?

File: lib/browser/fingerprint.ts [hunk 3]
```diff
@@ -72,8 +80,8 @@ export default function fingerprint(sdk: OktaAuthHttpInterface, options?: Finger
   return promise.finally(function() {
     clearTimeout(timeout);
     removeListener(window, 'message', listener);
-    if (document.body.contains(iframe)) {
-      iframe.parentElement.removeChild(iframe);
+    if (container.contains(iframe)) {
+      iframe.parentElement?.removeChild(iframe);
```
Comment (jaredperreault-okta) [inline lib/browser/fingerprint.ts:84]: What happens if `.parentElement.removeChild` is `undefined`. I assume it's possible since `?.` is being used

**Review by jaredperreault-okta — DISMISSED** [review]

**Review by jaredperreault-okta — APPROVED** [review]

**Pull request merged**